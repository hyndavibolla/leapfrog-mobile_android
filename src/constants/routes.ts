import { TransactionFilter } from '_models/offer';

export const ROUTES = {
  DEV_TOOLS: 'devTools',
  DEV_TOOLS_TAB: {
    LOG: 'devTools/logs',
    API_OVERRIDE: 'devTools/apiOverride',
    STORAGE: 'devTools/storageEditor',
    STORYBOOK: 'devTools/storybook'
  },
  SPLASH: 'splash',
  LOGIN: 'login',
  TERMS_AND_CONDITIONS: 'terms-and-conditions',
  PRIVACY_POLICY: 'privacy-policy',
  HOW_IT_WORKS: {
    TITLE: 'how-it-works',
    BENEFITS: 'how-it-works/benefits',
    LEVELS: 'how-it-works/levels',
    POINTS: 'how-it-works/points',
    FAQ: 'how-it-works/faqs'
  },
  MAIN: 'main',
  MAIN_TAB: {
    MAIN: 'main-tabs',
    STREAK: 'streaks',
    EARN: 'earn',
    REWARDS: 'rewards',
    WALLET: 'wallet'
  },
  WALLET: {
    REDEMPTIONS: 'redemptions',
    CREDIT_CARDS: 'credit-cards'
  },
  POINT_HISTORY: 'transactions',
  MISSION_SEE_ALL: 'earn/search',
  GIFT_CARD_SEE_ALL: 'rewards/search',
  MISSION_DETAIL: 'earn/offer',
  GIFT_CARD_LIST_DETAIL: 'wallet/giftcard/detail',
  GIFT_CARD_BALANCE: 'wallet/giftcard/balance',
  GIFT_CARD_GOOGLE_PAY: 'wallet/giftcard/google-pay',
  GIFT_CARD_DETAIL: 'rewards/brand',
  GIFT_CARD_CHECKOUT: 'rewards/gc/checkout',
  FUSION_VIEWER: 'cc-application-finished',
  PROFILE: 'profile',
  EDIT_PROFILE: 'profile/edit',
  DELETE_ACCOUNT: 'profile/deleteAccount',
  TOOLTIP: {
    ONBOARDING: 'onboarding/tooltip',
    ONBOARDING_STEP_1: 'onboarding/tooltip/1',
    ONBOARDING_STEP_2: 'onboarding/tooltip/2',
    ONBOARDING_STEP_3: 'onboarding/tooltip/3',
    EARN: 'earn/tooltip',
    EARN_STEP_1: 'earn/tooltip/1',
    EARN_STEP_2: 'earn/tooltip/2',
    REWARDS: 'rewards/tooltip',
    REWARDS_STEP_1: 'rewards/tooltip/1',
    REWARDS_STEP_2: 'rewards/tooltip/2',
    MISSIONS: 'missions/tooltip',
    MISSIONS_STEP_1: 'missions/tooltip/1',
    MISSIONS_STEP_2: 'missions/tooltip/2',
    MISSIONS_STEP_3: 'missions/tooltip/3'
  },
  TRACKING_MODALS: {
    EARN_BRAND_CATEGORY_LIST: 'earn/brand-category-list',
    REWARDS_CHOOSE_CATEGORIES: 'rewards/brand-category-list'
  },
  STREAK_LIST: 'streak/search',
  SURVEY_DETAIL: 'surveys',
  SURVEY_PQ: 'surveys/pq',
  IN_STORE_OFFERS: {
    OFFER_SEARCH_MAP: 'inStore/offer/map',
    MAP_SEARCH: 'inStore/map',
    OFFER_DETAIL: 'inStore/offer/detail',
    CARD_LINK: 'inStore/cardLink',
    WEB_VIEW: 'inStore/webview'
  },
  APPLY_NOW: 'apply-now',
  MANAGE_SYW_CARD: 'wallet/manage',
  WALLET_DETAIL: 'wallet/card/detail',
  FORCED_UPDATE_MODAL: 'forced-update',
  TRANSACTION_FILTERS: {
    MAIN: 'transaction/filters',
    TRANSACTIONS_LOCAL_OFFERS: `transaction/filters/${TransactionFilter.LOCAL_OFFERS}`,
    TRANSACTIONS_MISSIONS: `transaction/filters/${TransactionFilter.MISSIONS}`,
    TRANSACTIONS_REWARDS: `transaction/filters/${TransactionFilter.REWARDS}`,
    TRANSACTIONS_SYW_MASTERCARD: `transaction/filters/${TransactionFilter.SYW_MASTERCARD}`
  },
  COMMON_WEB_VIEW: {
    MAIN: 'webview'
  },
  FILTER: 'rewards/filter',
  MISSION_SELECT_CATEGORY: 'earn/select-category',
  REWARDS_LIST_BY_CATEGORY: 'rewards/search/category',
  WALLET_YOUR_GIFT_CARDS: {
    MAIN: 'wallet/your-gift-cards',
    TAB_YOUR_GIFT_CARDS_ACTIVE: 'your-gift-cards/active',
    TAB_YOUR_GIFT_CARDS_ARCHIVED: 'your-gift-cards/archived'
  },
  UNIFIMONEY: 'unifimoney'
};

export const ROUTES_PARAMS = {
  [ROUTES.TOOLTIP.ONBOARDING]: ['step'],
  [ROUTES.TOOLTIP.EARN]: ['step'],
  [ROUTES.TOOLTIP.REWARDS]: ['step'],
  [ROUTES.TOOLTIP.MISSIONS]: ['step'],
  [ROUTES.MISSION_DETAIL]: ['brandRequestorId'],
  [ROUTES.GIFT_CARD_DETAIL]: ['brandId'],
  [ROUTES.TRANSACTION_FILTERS.MAIN]: ['filterName']
};

export const ROUTES_CONFIG = {
  screens: {
    [ROUTES.DEV_TOOLS]: ROUTES.DEV_TOOLS,
    [ROUTES.DEV_TOOLS_TAB.LOG]: ROUTES.DEV_TOOLS_TAB.LOG,
    [ROUTES.DEV_TOOLS_TAB.API_OVERRIDE]: ROUTES.DEV_TOOLS_TAB.API_OVERRIDE,
    [ROUTES.DEV_TOOLS_TAB.STORAGE]: ROUTES.DEV_TOOLS_TAB.STORAGE,
    [ROUTES.DEV_TOOLS_TAB.STORYBOOK]: ROUTES.DEV_TOOLS_TAB.STORYBOOK,
    [ROUTES.SPLASH]: ROUTES.SPLASH,
    [ROUTES.LOGIN]: ROUTES.LOGIN,
    [ROUTES.TERMS_AND_CONDITIONS]: ROUTES.TERMS_AND_CONDITIONS,
    [ROUTES.PRIVACY_POLICY]: ROUTES.PRIVACY_POLICY,
    [ROUTES.HOW_IT_WORKS.TITLE]: {
      screens: {
        [ROUTES.HOW_IT_WORKS.TITLE]: ROUTES.HOW_IT_WORKS.TITLE,
        [ROUTES.HOW_IT_WORKS.BENEFITS]: ROUTES.HOW_IT_WORKS.BENEFITS,
        [ROUTES.HOW_IT_WORKS.LEVELS]: ROUTES.HOW_IT_WORKS.LEVELS,
        [ROUTES.HOW_IT_WORKS.POINTS]: ROUTES.HOW_IT_WORKS.POINTS,
        [ROUTES.HOW_IT_WORKS.FAQ]: ROUTES.HOW_IT_WORKS.FAQ
      }
    },
    [ROUTES.MAIN]: {
      screens: {
        [ROUTES.MAIN_TAB.MAIN]: {
          screens: {
            [ROUTES.MAIN_TAB.STREAK]: ROUTES.MAIN_TAB.STREAK,
            [ROUTES.MAIN_TAB.EARN]: ROUTES.MAIN_TAB.EARN,
            [ROUTES.MAIN_TAB.REWARDS]: ROUTES.MAIN_TAB.REWARDS,
            [ROUTES.MAIN_TAB.WALLET]: ROUTES.MAIN_TAB.WALLET
          }
        }
      }
    },
    [ROUTES.POINT_HISTORY]: ROUTES.POINT_HISTORY,
    [ROUTES.MISSION_SEE_ALL]: ROUTES.MISSION_SEE_ALL,
    [ROUTES.GIFT_CARD_SEE_ALL]: ROUTES.GIFT_CARD_SEE_ALL,
    [ROUTES.MISSION_DETAIL]: { path: `${ROUTES.MISSION_DETAIL}/:brandRequestorId?` },
    [ROUTES.GIFT_CARD_DETAIL]: { path: `${ROUTES.GIFT_CARD_DETAIL}/:brandId?` },
    [ROUTES.GIFT_CARD_BALANCE]: ROUTES.GIFT_CARD_BALANCE,
    [ROUTES.GIFT_CARD_GOOGLE_PAY]: ROUTES.GIFT_CARD_GOOGLE_PAY,
    [ROUTES.GIFT_CARD_CHECKOUT]: ROUTES.GIFT_CARD_CHECKOUT,
    [ROUTES.FUSION_VIEWER]: ROUTES.FUSION_VIEWER,
    [ROUTES.PROFILE]: ROUTES.PROFILE,
    [ROUTES.EDIT_PROFILE]: ROUTES.EDIT_PROFILE,
    [ROUTES.DELETE_ACCOUNT]: ROUTES.DELETE_ACCOUNT,
    [ROUTES.TOOLTIP.ONBOARDING]: { path: `${ROUTES.TOOLTIP.ONBOARDING}/:step?` },
    [ROUTES.TOOLTIP.EARN]: { path: `${ROUTES.TOOLTIP.EARN}/:step?` },
    [ROUTES.TOOLTIP.REWARDS]: { path: `${ROUTES.TOOLTIP.REWARDS}/:step?` },
    [ROUTES.TOOLTIP.MISSIONS]: { path: `${ROUTES.TOOLTIP.MISSIONS}/:step?` },
    [ROUTES.STREAK_LIST]: ROUTES.STREAK_LIST,
    [ROUTES.SURVEY_DETAIL]: ROUTES.SURVEY_DETAIL,
    [ROUTES.SURVEY_PQ]: ROUTES.SURVEY_PQ,
    [ROUTES.IN_STORE_OFFERS.OFFER_SEARCH_MAP]: ROUTES.IN_STORE_OFFERS.OFFER_SEARCH_MAP,
    [ROUTES.IN_STORE_OFFERS.MAP_SEARCH]: ROUTES.IN_STORE_OFFERS.MAP_SEARCH,
    [ROUTES.IN_STORE_OFFERS.OFFER_DETAIL]: ROUTES.IN_STORE_OFFERS.OFFER_DETAIL,
    [ROUTES.IN_STORE_OFFERS.CARD_LINK]: ROUTES.IN_STORE_OFFERS.CARD_LINK,
    [ROUTES.IN_STORE_OFFERS.WEB_VIEW]: ROUTES.IN_STORE_OFFERS.WEB_VIEW,
    [ROUTES.APPLY_NOW]: ROUTES.APPLY_NOW,
    [ROUTES.MANAGE_SYW_CARD]: ROUTES.MANAGE_SYW_CARD,
    [ROUTES.WALLET_DETAIL]: ROUTES.WALLET_DETAIL,
    [ROUTES.TRANSACTION_FILTERS.MAIN]: { path: `${ROUTES.TRANSACTION_FILTERS.MAIN}/:filterName?` }
  }
};

export const ARCHIVED_ROUTES = {
  'main/earn': ROUTES.MAIN_TAB.EARN,
  'main/rewards': ROUTES.MAIN_TAB.REWARDS,
  'main/wallet%20%3E%20cc': ROUTES.MAIN_TAB.WALLET,
  'How%20It%20Works': ROUTES.HOW_IT_WORKS.TITLE,
  'How%20It%20Works/how%20it%20works%20%3E%20benefits': ROUTES.HOW_IT_WORKS.BENEFITS,
  'How%20It%20Works/how%20it%20works%20%3E%20levels': ROUTES.HOW_IT_WORKS.LEVELS,
  'How%20It%20Works/how%20it%20works%20%3E%20points': ROUTES.HOW_IT_WORKS.POINTS,
  'How%20It%20Works/how%20it%20works%20%3E%20faqs': ROUTES.HOW_IT_WORKS.FAQ,
  Transactions: ROUTES.POINT_HISTORY,
  'earn%20%3E%20search%20results': ROUTES.MISSION_SEE_ALL,
  'rewards%20%3E%20search%20results': ROUTES.GIFT_CARD_SEE_ALL,
  ccApplicationFinished: ROUTES.FUSION_VIEWER,
  'earn%20tooltip': ROUTES.TOOLTIP.EARN,
  'rewards%20tooltip': ROUTES.TOOLTIP.REWARDS,
  'Terms%20and%20Conditions': ROUTES.TERMS_AND_CONDITIONS,
  'Privacy%20Policy': ROUTES.PRIVACY_POLICY,
  'rewards%20%3E%20gc%20checkout': ROUTES.GIFT_CARD_CHECKOUT,
  'Dev%20Tools': ROUTES.DEV_TOOLS,
  'Dev%20Tools/Logs': ROUTES.DEV_TOOLS_TAB.LOG,
  'Dev%20Tools/API%20Override': ROUTES.DEV_TOOLS_TAB.API_OVERRIDE,
  Splash: ROUTES.SPLASH,
  Login: ROUTES.LOGIN
};
