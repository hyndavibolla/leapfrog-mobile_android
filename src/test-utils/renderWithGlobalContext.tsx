import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
