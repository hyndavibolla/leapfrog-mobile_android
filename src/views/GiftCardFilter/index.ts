export { default as GiftCardFilter } from './GiftCardFilter';
