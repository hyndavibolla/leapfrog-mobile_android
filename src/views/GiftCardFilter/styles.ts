import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '_constants';

export const styles = {
  container: css`
    flex: 1;
    background-color: ${COLOR.LIGHT_GRAY};
    padding-top: 28px;
  `,
  headerContainer: css`
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    padding-horizontal: 16px;
  `,
  footerContainer: css`
    height: 100px;
    padding-horizontal: 32px;
    justify-content: center;
    background-color: ${COLOR.WHITE};
  `,
  title: css`
    font-size: ${FONT_SIZE.SMALLER};
    font-family: ${FONT_FAMILY.BOLD};
    color: ${COLOR.BLACK};
  `,
  clearText: css`
    font-size: ${FONT_SIZE.SMALLER};
    font-family: ${FONT_FAMILY.BOLD};
    color: ${COLOR.PRIMARY_BLUE};
  `,
  listContainer: css`
    flex: 1;
    margin-top: 20px;
  `,
  innerButton: css`
    padding: 16px 0;
  `,
  buttonText: css`
    font-family: ${FONT_FAMILY.HEAVY};
  `,
  listContent: css`
    padding-horizontal: 16px;
  `
};
