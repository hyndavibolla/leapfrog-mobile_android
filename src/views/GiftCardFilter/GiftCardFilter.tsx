import React, { memo, useState, useCallback, useContext } from 'react';
import { View, Text, TouchableOpacity, ScrollView } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';

import { Button } from '_components/Button';
import { SearchFilterItem } from '_commons/components/molecules/SearchFilterItem';

import { formatPrettyTitle } from '_utils/formatPrettyTitle';
import { useTestingHelper } from '_utils/useTestingHelper';

import { styles } from './styles';
import { ForterActionType, PageNames, PageType, TealiumEventType, UxObject } from '_constants/eventTracking';
import { useEventTracker } from '_state_mgmt/core/hooks';
import { GlobalContext } from '_state_mgmt/GlobalState';
import { ROUTES } from '_constants/routes';
import { useSetRecentSearchItem } from '_utils/useSetRecentSearchItem';
import { ENV } from '_constants/env';
import { IRecentSearchHistory, RecentSearchHistoryType } from '_models/searchHistory';

interface RouteParams {
  params: {
    categories: string[];
  };
}

export interface Props {
  route: RouteParams;
  navigation: StackNavigationProp<any>;
}

export const GiftCardFilter = ({
  navigation,
  route: {
    params: { categories }
  }
}: Props) => {
  const { getTestIdProps } = useTestingHelper('filter');
  const setRecentHistoryItem = useSetRecentSearchItem(ENV.STORAGE_KEY.GIFT_CARDS_SEARCH_HISTORY);
  const { state } = useContext(GlobalContext);
  const { trackUserEvent } = useEventTracker();
  const maxSelectedCategories = 1;
  const [innerCategorySelectionList, setInnerCategorySelectionList] = useState<string[]>([]);

  const onCategoryToggle = useCallback(
    (categoryId: string) =>
      setInnerCategorySelectionList(innerList => {
        if (innerList.includes(categoryId)) {
          return innerList.filter(c => c !== categoryId);
        } else {
          trackUserEvent(
            TealiumEventType.SELECT_GIFT_CARD_CATEGORY,
            {
              page_name: PageNames.REWARDS.REWARDS_CATEGORY,
              page_type: PageType.SELECTION,
              select_category: categoryId,
              event_type: categoryId,
              event_detail: categoryId,
              brand_category: categoryId,
              uxObject: UxObject.LIST
            },
            ForterActionType.TAP
          );
          return [...innerList, categoryId].slice(innerList.length + 1 - Math.min(maxSelectedCategories, innerList.length + 1));
        }
      }),
    [maxSelectedCategories, trackUserEvent]
  );

  const onClearAll = useCallback(() => {
    setInnerCategorySelectionList([]);
  }, []);

  const handleSubmit = useCallback(() => {
    const navigate = state.core.routeHistory[1] === ROUTES.REWARDS_LIST_BY_CATEGORY ? navigation.navigate : navigation.replace;
    const currentCategory = innerCategorySelectionList[0];
    if (currentCategory) {
      const historyItem: IRecentSearchHistory = { id: currentCategory, name: formatPrettyTitle(currentCategory), type: RecentSearchHistoryType.CATEGORY };
      setRecentHistoryItem(historyItem);
    }
    navigate(ROUTES.REWARDS_LIST_BY_CATEGORY, { searchCategoryIdList: innerCategorySelectionList });
  }, [navigation, state.core, innerCategorySelectionList, setRecentHistoryItem]);

  return (
    <View style={styles.container}>
      <View style={styles.headerContainer}>
        <Text style={styles.title}>Select a category</Text>
        <TouchableOpacity onPress={onClearAll} {...getTestIdProps('clear')}>
          <Text style={styles.clearText}>Clear</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.listContainer}>
        <ScrollView showsVerticalScrollIndicator={false} contentContainerStyle={styles.listContent}>
          {categories.map(categoryId => {
            const isSelected = innerCategorySelectionList.includes(categoryId);
            return (
              <SearchFilterItem
                key={categoryId}
                isSelected={isSelected}
                onPress={() => onCategoryToggle(categoryId)}
                identifier={categoryId}
                title={formatPrettyTitle(categoryId)}
              />
            );
          })}
        </ScrollView>

        <View style={styles.footerContainer}>
          <Button innerContainerStyle={styles.innerButton} textStyle={styles.buttonText} onPress={handleSubmit} {...getTestIdProps('apply')}>
            Apply
          </Button>
        </View>
      </View>
    </View>
  );
};

export default memo(GiftCardFilter);
