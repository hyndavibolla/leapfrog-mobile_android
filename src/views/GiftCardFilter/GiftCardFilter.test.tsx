import React from 'react';
import { act } from 'react-test-renderer';
import { fireEvent } from '@testing-library/react-native';

import GiftCardFilter, { Props } from './GiftCardFilter';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { getRewardConfig_1, getSlideCategoryIdList } from '_test_utils/entities';
import { wait } from '_utils/wait';
import { getInitialState } from '_state_mgmt/GlobalState';
import { IGlobalState } from '_models/general';
import { ROUTES } from '_constants/routes';

describe('GiftCardFilter', () => {
  let props: Props;
  let stateWithCategories: any;
  let initialState: IGlobalState;

  beforeEach(() => {
    initialState = getInitialState();
    stateWithCategories = { ...initialState, reward: { ...initialState.reward, config: getRewardConfig_1(), slideCategoryIdList: getSlideCategoryIdList() } };
    props = {
      navigation: { navigate: jest.fn(), replace: jest.fn() } as any,
      route: {
        params: {
          categories: getSlideCategoryIdList()
        }
      }
    };
  });

  it('should render', async () => {
    const { toJSON } = renderWithGlobalContext(<GiftCardFilter {...props} />, undefined, stateWithCategories);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should NOT render category items', async () => {
    const [, category_1] = getRewardConfig_1().categories.map(c => c.id);
    const { queryByTestId } = renderWithGlobalContext(<GiftCardFilter {...props} />, undefined, stateWithCategories);
    expect(queryByTestId(`filter-item-category-${category_1}-btn`)).toBeTruthy();
  });

  it('should render category list', async () => {
    const [, category_1] = getRewardConfig_1().categories.map(c => c.id);
    const { queryByTestId } = renderWithGlobalContext(<GiftCardFilter {...props} />);
    expect(queryByTestId(`filter-item-category-${category_1}-btn`)).toBeTruthy();
  });

  it('should navigate to search list with filters', async () => {
    const [, category_1] = getRewardConfig_1().categories.map(c => c.id);
    initialState.core.routeHistory = [ROUTES.FILTER, ROUTES.MAIN_TAB.REWARDS];
    const { getByTestId } = renderWithGlobalContext(<GiftCardFilter {...props} />, undefined, initialState);
    await act(() => wait(0));
    fireEvent.press(getByTestId(`filter-item-category-${category_1}-btn`));
    await act(() => wait(0));
    fireEvent.press(getByTestId('filter-apply'));
    await act(() => wait(0));
    expect(props.navigation.replace).toBeCalledWith(ROUTES.REWARDS_LIST_BY_CATEGORY, { searchCategoryIdList: [category_1] });
  });

  it('should navigate to search list without filters', async () => {
    const [, category_1] = getRewardConfig_1().categories.map(c => c.id);
    initialState.core.routeHistory = [ROUTES.FILTER, ROUTES.REWARDS_LIST_BY_CATEGORY];
    const { getByTestId } = renderWithGlobalContext(<GiftCardFilter {...props} />, undefined, initialState);
    await act(() => wait(0));
    fireEvent.press(getByTestId(`filter-item-category-${category_1}-btn`));
    await act(() => wait(0));
    fireEvent.press(getByTestId('filter-clear'));
    await act(() => wait(0));
    fireEvent.press(getByTestId('filter-apply'));
    await act(() => wait(0));
    expect(props.navigation.navigate).toBeCalledWith(ROUTES.REWARDS_LIST_BY_CATEGORY, { searchCategoryIdList: [] });
  });

  it('should navigate to search list without filters when unselect a category', async () => {
    const [, category_1] = getRewardConfig_1().categories.map(c => c.id);
    initialState.core.routeHistory = [ROUTES.FILTER, ROUTES.REWARDS_LIST_BY_CATEGORY];
    const { getByTestId } = renderWithGlobalContext(<GiftCardFilter {...props} />, undefined, initialState);
    await act(() => wait(0));
    fireEvent.press(getByTestId(`filter-item-category-${category_1}-btn`));
    await act(() => wait(0));
    fireEvent.press(getByTestId(`filter-item-category-${category_1}-btn`));
    await act(() => wait(0));
    fireEvent.press(getByTestId('filter-apply'));
    await act(() => wait(0));
    expect(props.navigation.navigate).toBeCalledWith(ROUTES.REWARDS_LIST_BY_CATEGORY, { searchCategoryIdList: [] });
  });
});
