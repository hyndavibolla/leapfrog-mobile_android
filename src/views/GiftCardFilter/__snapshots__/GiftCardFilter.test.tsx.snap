// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftCardFilter should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingTop": 28,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
        }
      }
    >
      Select a category
    </Text>
    <View
      accessibilityLabel="filter-clear"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="filter-clear"
    >
      <Text
        style={
          Object {
            "color": "#0066CC",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 16,
          }
        }
      >
        Clear
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingHorizontal": 16,
        }
      }
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          accessibilityLabel="filter-item-category-Entertainment-btn"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="filter-item-category-Entertainment-btn"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EBEDF2",
                  "borderBottomColor": "#EBEDF2",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderBottomWidth": 2,
                  "borderLeftColor": "#EBEDF2",
                  "borderLeftWidth": 2,
                  "borderRightColor": "#EBEDF2",
                  "borderRightWidth": 2,
                  "borderTopColor": "#EBEDF2",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
                false,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#262C2F",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  false,
                ]
              }
            >
               
              Entertainment
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="filter-item-category-Sports and Outdoors-btn"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="filter-item-category-Sports and Outdoors-btn"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EBEDF2",
                  "borderBottomColor": "#EBEDF2",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderBottomWidth": 2,
                  "borderLeftColor": "#EBEDF2",
                  "borderLeftWidth": 2,
                  "borderRightColor": "#EBEDF2",
                  "borderRightWidth": 2,
                  "borderTopColor": "#EBEDF2",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
                false,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#262C2F",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  false,
                ]
              }
            >
               
              Sports and Outdoors
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="filter-item-category-Travel-btn"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="filter-item-category-Travel-btn"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EBEDF2",
                  "borderBottomColor": "#EBEDF2",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderBottomWidth": 2,
                  "borderLeftColor": "#EBEDF2",
                  "borderLeftWidth": 2,
                  "borderRightColor": "#EBEDF2",
                  "borderRightWidth": 2,
                  "borderTopColor": "#EBEDF2",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
                false,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#262C2F",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  false,
                ]
              }
            >
               
              Travel
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="filter-item-category-Automotive-btn"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="filter-item-category-Automotive-btn"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EBEDF2",
                  "borderBottomColor": "#EBEDF2",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderBottomWidth": 2,
                  "borderLeftColor": "#EBEDF2",
                  "borderLeftWidth": 2,
                  "borderRightColor": "#EBEDF2",
                  "borderRightWidth": 2,
                  "borderTopColor": "#EBEDF2",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderTopWidth": 2,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
                false,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#262C2F",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  false,
                ]
              }
            >
               
              Automotive
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "height": 100,
          "justifyContent": "center",
          "paddingHorizontal": 32,
        }
      }
    >
      <View
        accessibilityLabel="filter-apply"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="filter-apply"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#0066CC",
                "borderBottomLeftRadius": 30,
                "borderBottomRightRadius": 30,
                "borderTopLeftRadius": 30,
                "borderTopRightRadius": 30,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
              },
              Object {
                "paddingBottom": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 16,
              },
            ]
          }
        >
          <View
            style={false}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontFamily": "SFProDisplay-Medium",
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "SFProDisplay-Heavy",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Apply
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
